src/plugins/map/MQ2MapCommands.cpp

```diff
@@ -1169,151 +1182,350 @@ char* FormatMarker(const char* szLine, char* szDest, size_t BufferSize)
+		[](MapFilterOption* a, MapFilterOption* b) -> bool
+		{
+			if (a == nullptr || a->szName == nullptr) return false;
+			if (b == nullptr || b->szName == nullptr) return true;
+			return ci_less()(a->szName, b->szName);
+		}
+	);
+
+	// Build filtered arrays by type
+	for (auto option : allFilterOptions)
+	{
+		if (option->IsToggle() || option->IsRadius())
+		{
+			if (option->IsObject())
+			{
+				mapfilterObjectOptions.push_back(option);
+			}
+			else
+			{
+				mapFilterOptions.push_back(option);
+			}
+		}
+	}
 
+	// Put children under parents
+	BuildFilteredOptionArray(mapfilterObjectOptions);
+	BuildFilteredOptionArray(mapFilterOptions);
+}
+
+/// <summary>
+/// Adds ImGui setting object
+/// </summary>
+/// <returns>true if needs to regenerate</returns>
+static bool AddMapFilterOptionAsImGuiSetting(MapFilterOption* option, std::stack<MapFilterOption*> &optionStack)
+{
+	bool changed = false;
+	bool regenerate = false;
+	bool isRequirementMet = RequirementsMet(option->ThisFilter);
+
+	ImGui::PushID(option);
+
+	// if this requires previous, indent
+	// if this req is equal to previous req, no change
+	// if this requires something new, not previous, pop-loop (and unindent) until previous conditions are met or until stack empty
+	MapFilterOption* previousOption;
+	bool nested = false;
+	while (optionStack.size() > 0)
+	{
+		previousOption = optionStack.top();
+		if (previousOption->ThisFilter == option->RequiresOption)
+		{
+			ImGui::Indent();
+			nested = true;
+			break;
+		}
+		else if (previousOption->RequiresOption == option->RequiresOption)
+		{
+			optionStack.pop();
+			nested = true;
+			break;
+		}
+		else
+		{
+			optionStack.pop();
+			if (optionStack.size() < 1)
+				break;
+			ImGui::Unindent();
+		}
+	}
+
+	optionStack.push(option);
+
+	if (!isRequirementMet)
+		ImGui::BeginDisabled();
+
+	std::string radiusName = "Radius";
+	if (option->IsToggle())
+	{
+		if (ImGui::Checkbox(option->szName, &option->Enabled))
+			changed = true;
+	}
+	if (option->IsRadius())
+	{
+		if (option->IsToggle())
+			ImGui::SameLine();
+		else
+			radiusName = option->szName;
+
+		ImGui::SetNextItemWidth(40);
+		if (ImGui::InputFloat(radiusName.c_str(), &option->Radius, 0.0f, 0.0f, "%.0f"))
+		{
+			if (option->Radius < 0)
+				option->Radius = 0;
+			changed = true;
+		}
+	}
+
+	// Draw a tooltip (?) if this option has requirements from another option list
+	MapFilterOption& requirement = GetMapFilterOption(option->RequiresOption);
+	if (!nested && requirement.szName != nullptr && requirement.ThisFilter != MapFilter::All)
+	{
+		ImGui::SameLine();
+
+		mq::imgui::HelpMarker(
+			[&]() -> const char*
+			{
+				std::string requireString = "Requires: ";
+				if (requirement.IsObject() != option->IsObject())
+					requireString += requirement.IsObject() ? "Object Filters -> " : "Options -> ";
+
+				requireString += std::string(requirement.szName);
+				char* requireArray = &requireString[0];
```

Comment:
This is returning a pointer to a temporary that is out of scope.

std::requireString no longer exists when the lambda function returns, so using the pointer will be undefined behavior.

Reply:
this was our last solution to removing the memory leak.  Do I just need to assign it first or something?